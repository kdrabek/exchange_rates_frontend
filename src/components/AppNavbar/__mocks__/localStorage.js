export default class LocalStorageMock {
  constructor() {
    this.storage = {};
  }

  getItem(key) {
    return this.storage[key];
  }

  setItem(key, value) {
    this.storage[key] = value.toString();
  }

  clear() {
    this.storage = {};
  }

};