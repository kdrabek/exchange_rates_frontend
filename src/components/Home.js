import React from 'react';
import { Link } from 'react-router';
import { Column, Colors, Row, Link as ButtonLink } from 'react-foundation';

class Home extends React.Component {
  render() {
    return (
      <Row>
        <Column large={3}>
          <h4 className="subheader">Opcje</h4>
          <hr/>
          <p>
            Wybierz datę:
            <input type="date" name="bday" min="2016-12-01" value="2016-02-08" />
          </p>
          <Link to="notifications">
            <ButtonLink color={Colors.PRIMARY}>Przejdź do notyfikacji</ButtonLink>
          </Link>
        </Column>
        <Column large={9}>
          <h4 className="subheader">Akutalne kursy walut</h4>
          <hr/>
          <table class="unstriped hover">
            <thead>
              <tr>
                <th width="200">Kraj</th>
                <th>Waluta</th>
                <th width="150">Symbol</th>
                <th width="150">Kurs</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td><img src="../../public/img/AUD.png" /> Australia</td>
                <td>Dolar australijski</td>
                <td>AUD</td>
                <td>4.5678</td>
              </tr>
              <tr>
                <td><img src="../../public/img/AUD.png" /> Australia</td>
                <td>Dolar australijski</td>
                <td>AUD</td>
                <td>4.5678</td>
              </tr>
              <tr>
                <td><img src="../../public/img/AUD.png" /> Australia</td>
                <td>Dolar australijski</td>
                <td>AUD</td>
                <td>4.5678</td>
              </tr>
            </tbody>
          </table>

        </Column>
      </Row>
    );
  }
}

export default Home;
