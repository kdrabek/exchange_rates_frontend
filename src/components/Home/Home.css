.force-text-left {
    text-align: left;
}