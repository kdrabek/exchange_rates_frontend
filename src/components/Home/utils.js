import AUD from '../../../public/img/AUD.png';
import BGN from '../../../public/img/BGN.png';
import BRL from '../../../public/img/BRL.png';
import CAD from '../../../public/img/CAD.png';
import CHF from '../../../public/img/CHF.png';
import CLP from '../../../public/img/CLP.png';
import CNY from '../../../public/img/CNY.png';
import CZK from '../../../public/img/CZK.png';
import DKK from '../../../public/img/DKK.png';
import EUR from '../../../public/img/EUR.png';
import GBP from '../../../public/img/GBP.png';
import HKD from '../../../public/img/HKD.png';
import HRK from '../../../public/img/HRK.png';
import HUF from '../../../public/img/HUF.png';
import IDR from '../../../public/img/IDR.png';
import ILS from '../../../public/img/ILS.png';
import INR from '../../../public/img/INR.png';
import ISK from '../../../public/img/ISK.png';
import JPY from '../../../public/img/JPY.png';
import KRW from '../../../public/img/KRW.png';
import MXN from '../../../public/img/MXN.png';
import MYR from '../../../public/img/MYR.png';
import NOK from '../../../public/img/NOK.png';
import NZD from '../../../public/img/NZD.png';
import PHP from '../../../public/img/PHP.png';
import PLN from '../../../public/img/PLN.png';
import RON from '../../../public/img/RON.png';
import RUB from '../../../public/img/RUB.png';
import SEK from '../../../public/img/SEK.png';
import SGD from '../../../public/img/SGD.png';
import THB from '../../../public/img/THB.png';
import TRY from '../../../public/img/TRY.png';
import UAH from '../../../public/img/UAH.png';
import USD from '../../../public/img/USD.png';
import XDR from '../../../public/img/XDR.png';
import ZAR from '../../../public/img/ZAR.png';


export const currencyCodePictures = {
  AUD,
  BGN,
  BRL,
  CAD,
  CHF,
  CLP,
  CNY,
  CZK,
  DKK,
  EUR,
  GBP,
  HKD,
  HRK,
  HUF,
  IDR,
  ILS,
  INR,
  ISK,
  JPY,
  KRW,
  MXN,
  MYR,
  NOK,
  NZD,
  PHP,
  PLN,
  RON,
  RUB,
  SEK,
  SGD,
  THB,
  TRY,
  UAH,
  USD,
  XDR,
  ZAR
};

export const plDayLabels = ['Nie', 'Pon', 'Wt', 'Śr', 'Czw', 'Pt', 'Sob'];
export const plMonthLabels = [
  'Styczeń', 'Luty', 'Marzec', 'Kwiecień', 'Maj', 'Czerwiec', 'Lipiec',
  'Sierpień', 'Wrzesień', 'Październik', 'Listopad', 'Grudzień'
];

