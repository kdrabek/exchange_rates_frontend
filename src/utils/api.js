const BASE_API_URL = 'http://localhost';

class Api {
  
  getCurrentRates() {
    const url = '/rates/rates';
    return fetch(`${BASE_API_URL}${url}`).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }

  getRatesForDate(date) {
    const url = `/rates/rates/${date}`;
    return fetch(`${BASE_API_URL}${url}`).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }

  getRatesForCurrency(token, currency, days) {
    const url = `/rates/rates/${currency}/limit/${days}`;
    let headers = new Headers();
    headers.append('Authorization', `Token ${token}`);

    return fetch(
      `${BASE_API_URL}${url}`,
      {
        method: 'GET',
        headers: headers,
        mode: 'cors'
      }
    ).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }

  getNotifications(token) {
    const url = `/notifications/${token}`;
    let headers = new Headers();
    headers.append('Authorization', `Token ${token}`);

    return fetch(
      `${BASE_API_URL}${url}`,
      {
        method: 'GET',
        headers: headers,
        mode: 'cors'
      }
    ).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }
  
  deleteNotification(token, notificationId) {
    const url = `/notifications/${token}/${notificationId}`;
    let headers = new Headers();
    headers.append('Authorization', `Token ${token}`);

    return fetch(
      `${BASE_API_URL}${url}`,
      {
        method: 'DELETE',
        headers: headers,
        mode: 'cors'
      }
    ).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }

  updateNotification(token, notification) {
    const url = `/notifications/${token}/${notification.id}`;
    let headers = new Headers();
    headers.append('Authorization', `Token ${token}`);

    return fetch(
      `${BASE_API_URL}${url}`,
      {
        method: 'PUT',
        headers: headers,
        mode: 'cors',
        body: JSON.stringify(notification)
      }
    ).then(response => {
      return response.json();
    }).catch(err => {return err;});
  }

  login(user) {
    const url = '/auth/login';
    return fetch(`${BASE_API_URL}${url}`, {
      method: 'POST',
      body: JSON.stringify(user)
    })
    .then(response => {
      return response.json().then(json => {
      if (!response.ok)
        return Promise.reject(json);
      return json;
      })
    })
  }
  
  register(user) {
    const url = '/auth/register';
    return fetch(`${BASE_API_URL}${url}`, {
      method: 'POST',
      body: JSON.stringify(user)
    })
    .then(response => {
      return response.json().then(json => {
      if (!response.ok)
        return Promise.reject(json);
      return json;
      })
    })
  }
}

export default Api;
