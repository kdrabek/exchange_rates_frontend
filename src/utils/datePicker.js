export const plDayLabels = ['Nie', 'Pon', 'Wt', 'Śr', 'Czw', 'Pt', 'Sob'];
export const plMonthLabels = [
  'Styczeń', 'Luty', 'Marzec', 'Kwiecień', 'Maj', 'Czerwiec', 'Lipiec',
  'Sierpień', 'Wrzesień', 'Październik', 'Listopad', 'Grudzień'
];

