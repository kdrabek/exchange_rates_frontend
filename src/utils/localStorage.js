export class LocalStorageMock {
  constructor() {
    this.store = {};
  }

  clear() {
    this.store = {};
  }

  getItem(key) {
    return this.store[key];
  }

  setItem(key, value) {
    this.store[key] = value.toString();
  }

  removeItem(key) {
    this.store[key] = undefined;
  }
};

export const setUserInfo = (token, email) => {
  localStorage.setItem('AuthUserToken', token);
  localStorage.setItem('AuthUserEmail', email);
};

export const removeUserInfo = () => {
  localStorage.removeItem('AuthUserToken');
  localStorage.removeItem('AuthUserEmail');
};

export const isAuthenticated = () => {
  return (
    localStorage.getItem('AuthUserToken') && localStorage.getItem('AuthUserEmail')
  );
}