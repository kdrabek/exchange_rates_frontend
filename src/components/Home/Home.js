import React, { Component } from 'react';
import { Link } from 'react-router';
import { Row, Col, FormGroup, ControlLabel, Table, Button } from 'react-bootstrap';
import  DatePicker  from 'react-bootstrap-date-picker';
import { connect } from 'react-redux';
import moment from 'moment';

import * as ratesActions from '../../actions/actions';
import { currencyCodePictures } from '../../utils/currencyCodes';
import { plDayLabels, plMonthLabels } from '../../utils/plLocale';
import RatesTable from '../RatesTable/RatesTable';

class Home extends Component {

  componentDidMount() {
    this.props.dispatch(ratesActions.loadRates());
  }

  handleChange(date) {
    const formattedDate = moment(date).format('YYYY-MM-DD');
    this.props.dispatch(ratesActions.loadRatesForDate(formattedDate));
  }

  notificationsButton() {
    return (
      <Link to="/notifications">
        <Button bsStyle="primary" block>Przejdź do notyfikacji</Button>
      </Link>
    );
  }

  prepareTableData(){
    if (!this.props.rates.rates) {
      return [['Downloading...']];
    } else {
      return this.props.rates.rates.map((rate, index) => {
        const linkTo = `/details/${rate.currency}`;
        return [
          <img src={currencyCodePictures[rate.currency]} alt={rate.currency}/>,
          rate.country,
          rate.name,
          rate.currency,
          rate.rate,
          <Link to={linkTo}>Zobacz więcej...</Link>
        ];
      });
    }
  }

  render() {
    const isAuthenticated = (
      localStorage.getItem('AuthUserToken') !== null &&
      localStorage.getItem('AuthUserEmail') !== null
    );
    const notificationBtn = isAuthenticated ? this.notificationsButton() : '';
    return (
    <Row>
      <Col xs={3} md={3}>
        <h3 className="force-text-left">Opcje</h3>
        <hr />
        <FormGroup>
          <ControlLabel>Wybierz datę: </ControlLabel>
          <DatePicker
            id="example-datepicker"
            dateFormat="YYYY/MM/DD"
            weekStartsOnMonday
            dayLabels={plDayLabels}
            monthLabels={plMonthLabels}
            onChange={this.handleChange.bind(this)}
            value={this.props.rates.tableDate}
          />
        </FormGroup>
        {notificationBtn}
      </Col>
      <Col xs={9} md={9}>
        <h3>Kursy walut ({this.props.rates.tableDate})</h3>
        <hr />
        <RatesTable 
          headers={[' ', 'Kraj', 'Waluta', 'Symbol', 'Kurs', ' ']} 
          data={this.prepareTableData()}
        />

      </Col>
    </Row>
    );
  }
}

function mapStateToProps(state, ownProps) {
  return {
    rates: state.rates,
    tableDate: state.tableDate,
    user: state.user
  };
}

export default connect(mapStateToProps)(Home);
