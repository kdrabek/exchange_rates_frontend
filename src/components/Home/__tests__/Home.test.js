import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { Home } from '../Home';
import { LocalStorageMock, removeUserInfo } from '../../../utils/localStorage';
global.localStorage = new LocalStorageMock;

describe('Home component', () => {

  beforeEach(()=>{
    removeUserInfo();
  });

  it('matches the snapshot', () => {
    const rates = {
      tableDate: '2017-04-11',
      rates:[{country: "XXX", currency: "XYZ", name: "name", rate: "1.23"}]
    };
    
    const tree = renderer.create(
      <Home 
        rates={rates} 
        tableDate="2017-04-11"
        loadRates={() => {}}
        loadRatesForDate={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

})
