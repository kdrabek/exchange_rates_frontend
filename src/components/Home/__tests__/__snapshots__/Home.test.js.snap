exports[`Home component matches the snapshot 1`] = `
<div
  className="row">
  <div
    className="col-md-3 col-xs-3">
    <div>
      <h3>
        Opcje
      </h3>
      <hr />
      <div
        className="form-group">
        <label
          className="control-label"
          htmlFor={undefined}>
          Wybierz datę:
        </label>
        <span
          className="input-group"
          id="example-datepicker_group">
          <div
            style={
              Object {
                "position": "relative",
              }
            } />
          <input
            data-formattedvalue="2017/04/11"
            id="example-datepicker"
            name={undefined}
            type="hidden"
            value="2017-04-11T12:00:00.000Z" />
          <input
            autoFocus={false}
            className="form-control"
            disabled={false}
            id={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="YYYY/MM/DD"
            style={
              Object {
                "width": "100%",
              }
            }
            type="text"
            value="2017/04/11" />
          <span
            className="input-group-addon"
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }>
            <div
              style={
                Object {
                  "opacity": 1,
                }
              }>
              ×
            </div>
          </span>
        </span>
      </div>
      
    </div>
  </div>
  <div
    className="col-md-9 col-xs-9">
    <h3>
      Kursy walut (
      2017-04-11
      )
    </h3>
    <hr />
    <table
      className="table table-striped table-hover">
      <thead>
        <tr>
          <th>
             
          </th>
          <th>
            Kraj
          </th>
          <th>
            Waluta
          </th>
          <th>
            Symbol
          </th>
          <th>
            Kurs
          </th>
          <th>
             
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <img
              alt="XYZ"
              src={undefined} />
          </td>
          <td>
            XXX
          </td>
          <td>
            name
          </td>
          <td>
            XYZ
          </td>
          <td>
            1.23
          </td>
          <td>
            <a
              onClick={[Function]}
              style={Object {}}>
              Zobacz więcej...
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
